### PR TITLE
Remove "experimental" wording from specs

### PR DESCRIPTION
**Pull request by azdagron**
Signed-off-by: Andrew Harding <aharding@vmware.com>

### CODE REVIEW

File: standards/SPIFFE_Workload_API.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Status of this Memo
 
-This document specifies an experimental identity API standard for the internet community, and requests discussion and suggestions for improvements. It is a work in progress. Distribution of this document is unlimited.
+This document specifies an identity API standard for the internet community, and requests discussion and suggestions for improvements. It is a work in progress. Distribution of this document is unlimited.
```
Comment (spikecurtis) [inline standards/SPIFFE_Workload_API.md]: Here and in other specs, we want to keep the "work in progress" language?

Comment (azdagron) [reply standards/SPIFFE_Workload_API.md]: Oh that's a good question. Does "work in progress" imply unstable? Does lack of "work in progress" mean "won't change"?

My two cents is that the specs are fairly stable, and any revisions would have compatibility in mind, and as such, we could probably remove that language as well.

Comment (azdagron) [reply standards/SPIFFE_Workload_API.md]: I'm actually wondering if we can just drop the whole "Status of this Memo" section from each document....

Comment (spikecurtis) [reply standards/SPIFFE_Workload_API.md]: I think we're at a place, at least for most of the specs (maybe we should discuss each individually?), where we consider them _stable_.  This doesn't mean they will not change at all, but that any changes will take back compatibility very carefully into consideration.

I think "work in progress" implies unstable, and we should explicitly switch to language that talks about stability.  Keeping the "status of this memo" makes sense in this regard, and anticipates a future where one or more specs get deprecated in favor of new stuff.

**Review by spikecurtis — APPROVED** [review]

**Pull request merged**